server/v2/server.go

```diff
@@ -23,12 +24,14 @@ type ServerComponent[T transaction.Tx] interface {
 
 	Start(context.Context) error
 	Stop(context.Context) error
-	Init(AppI[T], *viper.Viper, log.Logger) error
+	Init(servercore.AppI[T], *viper.Viper, log.Logger) error
 }
 
 // HasCLICommands is a server module that has CLI commands.
 type HasCLICommands interface {
-	CLICommands() CLIConfig
+	GetCommands(servercore.AppCreator[transaction.Tx]) []*cobra.Command
```

Comment:
Abstracted `HasCLICommands` to get each `Command`, `Queries`, `Txs` of a server module.
```go
type HasCLICommands interface {
	GetCommands(servercore.AppCreator[transaction.Tx]) []*cobra.Command
	GetTxs() []*cobra.Command
	GetQueries() []*cobra.Command
}
```
`CLICommands` will sum up all their commands as `CLIConfig`.
cc @julienrbrt @tac0turtle